File: selection_sort/selection_sort.c
#include <stdio.h>

void algoritmo(int lista[],int longitud);

int main()
{
    int lista[] = {9,0,10,8,1,3,12,-10,8,5,7};
    int longitud = sizeof(lista) / sizeof(lista[0]);
    algoritmo(lista,longitud);
    for (int i =0; i<longitud;i++)
    {
        printf("%i\n",lista[i]);
    }
    return 0;
}

void algoritmo(int lista[],int longitud)
{
    //numero de indice de inicio y ciclos
    int indice = 0;
    while (indice<longitud)
    {
        //comparaciones
        for (int indice_max = 1 + indice; indice_max<longitud;indice_max++)
        {
            if (lista[indice] > lista[indice_max])
            {
                //cambiando de lugar
                int num_mayor = lista[indice];
                *(lista+indice) = lista[indice_max];
                *(lista+indice_max) = num_mayor;
            }
        }
        indice ++;
    }
}

File: selection_sort/selection_sort.py
def algoritmo(lista:list):
    #indice minimo y ciclos
    for indice in range(0,len(lista)):
        #comparaciones
        for indice_max in range(indice+1,len(lista)):
            if lista[indice] > lista[indice_max]:
                num_mayor = lista[indice]
                lista[indice] = lista[indice_max]
                lista[indice_max] = num_mayor
    return lista
    
lista = [0,11,9,-2,7,6,4,5,20,0]

print(algoritmo(lista))

#cambia la lista hasta que encontro un numero mas pequeño que el de el indice minimo